x-pack/plugins/fleet/server/services/epm/packages/_install_package.ts

```diff
@@ -46,147 +47,160 @@ export async function _installPackage({
   installSource: InstallSource;
 }): Promise<AssetReference[]> {
   const { name: pkgName, version: pkgVersion } = packageInfo;
```

Comment:
All the changes in this file are from wrapping everything below in a `try - catch` block and handling the errors at the end.

Reply:
Is it possible to catch the error here instead of wrapping _installPackage?
https://github.com/elastic/kibana/blob/master/x-pack/plugins/fleet/server/routes/epm/handlers.ts#L201 .   I understand you'd have to do it twice for upload and registry but seems like it'd be neater.

Reply:
I found it best to catch the saved object conflict as close as possible to where it's happening, to be able to add a specific diagnostic error message to it. Also, there are more code paths that end up in `_installPackage()`, for example the bulk installation endpoint and `/setup`.

The error that is then thrown, `ConcurrentInstallOperationError`, is handled like all our other errors, and gets its `409` status from `getHTTPResponseCode()` in `server/errors/handlers`, no matter through what endpoint it was triggered.